File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
    vector<int> cnt(26,INT_MAX);
    vector<string> res;
    for(auto word:words){
        vector<int> cnt1(26,0);
        for(auto c:word){
            cnt1[c-'a']++;
        }
        for(int i=0;i<26;i++){
            cnt[i]=min(cnt1[i],cnt[i]);
        }
    }
    for(int i=0;i<26;i++){
        for(int j=0;j<cnt[i];j++){
            res.push_back(string(1,i+'a'));
        }
    }
    return res;    
}
};

File: 1002-find-common-characters/1002-find-common-characters.java
class Solution {
    	public  List<String> commonChars(String[] words) {
	    int charcount[]=new int[26];
	    Arrays.fill(charcount,Integer.MAX_VALUE);
	    for(String str:words){
	        int count[]=buildCharFrequencyTable(str);
	    for(int i=0;i<26;i++){
	        charcount[i]=Math.min(charcount[i],count[i]);
	    }
	    }
	    return toList(charcount);
    }
    
private  int[] buildCharFrequencyTable(String str) {
    int[] count = new int[26];
    for (char c : str.toCharArray()) {
        count[c - 'a']++;
    }
    return count;
}

private  List<String> toList(int[] charcount) {
    List<String> result = new ArrayList<>();
    for (int i = 0; i < charcount.length; i++) {
        for (int sz = charcount[i]; sz > 0; sz--) {
            result.add("" + (char) (i + 'a'));
        }
    }
    return result;
}
}